src/Book.ts

```diff
@@ -0,0 +1,197 @@
+        timestamp: "$timestamp",
+      },
+    };
+    const group: PipelineStage.Group = {
+      $group: {
+        // https://github.com/Automattic/mongoose/pull/11104
+        _id: null as any,
+        credit: {
+          $sum: "$credit",
+        },
+        debit: {
+          $sum: "$debit",
+        },
+        count: {
+          $sum: 1,
+        },
+      },
+    };
+    let result;
+    if (typeof skip !== "undefined") {
+      const sort: PipelineStage.Sort = {
+        $sort: {
+          datetime: -1,
+          timestamp: -1,
+        },
+      };
+      result = (
+        await transactionModel
+          .aggregate([match, project, sort, skip, group], options)
+          .exec()
+      )[0];
+    } else {
+      result = (
+        await transactionModel
+          .aggregate([match, project, group], options)
+          .exec()
+      )[0];
+    }
+    return !result
+      ? {
+          balance: 0,
+          notes: 0,
+        }
+      : {
+          balance: parseFloat(
+            (result.credit - result.debit).toFixed(this.precision)
+          ),
+          notes: result.count,
+        };
+  }
+
+  async ledger<T = U>(
+    query: IParseQuery & IPaginationQuery,
+    populate?: string[] | null,
+    options?: IOptions & { lean?: true }
+  ): Promise<{ results: T[]; total: number }>;
+
+  async ledger<T = U>(
+    query: IParseQuery & IPaginationQuery,
+    populate?: string[] | null,
+    options?: IOptions & { lean?: false }
+  ): Promise<{ results: (Document & T)[]; total: number }>;
+
+  async ledger<T = U>(
+    query: IParseQuery & IPaginationQuery,
+    populate = null as string[] | null,
+    options = {} as IOptions & { lean?: boolean }
+  ): Promise<{ results: T[]; total: number }> {
+    let skip;
+    let limit = 0;
+
+    const { lean = true } = options;
+
+    // Pagination
+    if (query.perPage) {
+      skip = (query.page ? query.page - 1 : 0) * query.perPage;
+      limit = query.perPage;
+    }
+    const filterQuery = parseQuery(query, { name: this.name });
+    const q = transactionModel
+      .find(filterQuery, undefined, options)
+      .lean(lean)
+      .sort({
+        datetime: -1,
+        timestamp: -1,
+      });
+
+    let count = Promise.resolve(0);
+    if (typeof skip !== "undefined") {
+      count = transactionModel
+        .countDocuments(filterQuery)
+        .session(options.session || null)
+        .exec();
+      q.skip(skip).limit(limit);
+    }
+
+    if (populate) {
+      for (let i = 0, il = populate.length; i < il; i++) {
+        if (isValidTransactionKey<U>(populate[i])) {
+          q.populate(populate[i]);
+        }
+      }
+    }
+
+    const results = (await q.exec()) as unknown as T[];
+
+    return {
+      results,
+      total: (await count) || results.length,
+    };
+  }
+
+  async void(
+    journal_id: string,
+    reason?: undefined | string,
+    options = {} as IOptions
+  ) {
+    const journal: TJournalDocument<J> = (await journalModel
```

Comment:
Maybe we should throw an Error if we dont find the requested Journal?

Reply:
Now it is book-scoped and throws an error if we dont find thejournal within the book